projects/storefrontlib/cms-components/myaccount/new-consent-management/components/new-consent-management.component.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  AnonymousConsentsConfig,
+  AnonymousConsentsService,
+  AuthService,
+  ConsentTemplate,
+  GlobalMessageService,
+  GlobalMessageType,
+  UserConsentService,
+} from '@spartacus/core';
+import {
+  BehaviorSubject,
+  combineLatest,
+  concat,
+  Observable,
+  Subscription,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  scan,
+  skipWhile,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { ConsentManagementComponentService } from '../../consent-management/consent-management-component.service';
+
+@Component({
+  selector: 'cx-new-consent-management',
+  templateUrl: './new-consent-management.component.html',
+})
+export class NewConsentManagementComponent implements OnInit, OnDestroy {
+  private subscriptions = new Subscription();
+  private allConsentsLoading = new BehaviorSubject<boolean>(false);
+
+  templateList$: Observable<ConsentTemplate[]>;
+  loading$: Observable<boolean>;
+
+  requiredConsents: string[] = [];
+
+  constructor(
```

Comment:
While it hasn't triggered sonar yet, any additional service in this constructor will which is going to make it difficult if we need to add to or deprecate the constructor of this component. This class is also very large and complicated. Can we consider breaking down this component and/or moving some of the logic to a service?

Reply:
Yeah, agree. But this file is copied from consent-management.component.ts without any changes. The new component will replace the existing one later. Let's keep the code structure as it is. It will make the replacement easier. We can create a ticket for it to enhance the code.